### ISSUE TITLE
create_certs.sh requires JAVA_HOME

### ISSUE THREAD

**Issue by boaks** [eclipse/hono#3246]
Commit: 4efa2465802cb139883054568c32d04bdc60dee8

Building with

```
$> mvn clean install
```

I get

> INFO] Running org.eclipse.hono.util.CredentialsObjectTest
> [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.001 s <<< FAILURE! - in org.eclipse.hono.util.TenantObjectTest
> [ERROR] org.eclipse.hono.util.TenantObjectTest  Time elapsed: 0.001 s  <<< ERROR!
> java.io.FileNotFoundException: target/certs/trustStore.jks (Datei oder Verzeichnis nicht gefunden)
> 	at org.eclipse.hono.util.TenantObjectTest.getCaCertificate(TenantObjectTest.java:68)
> 	at org.eclipse.hono.util.TenantObjectTest.init(TenantObjectTest.java:63)
> ...
> 08:20:34.301 [ForkJoinPool-1-worker-3] WARN  o.e.hono.config.AbstractConfigTest$1 - unsupported key & cert format: PKCS12
> 08:20:34.307 [ForkJoinPool-1-worker-3] WARN  o.e.hono.config.AbstractConfigTest$1 - unsupported key store format: PEM
> 08:20:34.319 [ForkJoinPool-1-worker-3] WARN  o.e.hono.config.AbstractConfigTest$1 - cannot read trust store password, file [/non-existing-file] does not exist
> 08:20:34.323 [ForkJoinPool-1-worker-3] WARN  o.e.hono.config.AbstractConfigTest$1 - cannot read key store password, file [/non-existing-file] does not exist
> 
> 

Content of my "core/target/certs":

```
amqp-adapter-cert.pem
amqp-adapter-key.pem
amqpKeyStore.p12
artemis-cert.pem
artemis-key.pem
artemisKeyStore.p12
auth-server-cert.pem
auth-server-key.pem
authServerKeyStore.p12
ca-cert.pem
ca-cert.srl
ca-key.pem
coap-adapter-cert.pem
coap-adapter-key.pem
coapKeyStore.p12
command-router-cert.pem
command-router-key.pem
commandRouterKeyStore.p12
default_tenant-cert.pem
default_tenant-cert.srl
default_tenant-key.pem
device-4711-cert.pem
device-4711-key.pem
device-certs.properties
device-registry-cert.pem
device-registry-key.pem
deviceRegistryKeyStore.p12
example-gateway-cert.pem
example-gateway-key.pem
exampleGatewayKeyStore.p12
http-adapter-cert.pem
http-adapter-key.pem
httpKeyStore.p12
lora-adapter-cert.pem
lora-adapter-key.pem
loraKeyStore.p12
mqtt-adapter-cert.pem
mqtt-adapter-key.pem
mqttKeyStore.p12
qdrouter-cert.pem
qdrouter-key.pem
root-cert.pem
root-cert.srl
root-key.pem
trust-anchor.properties
trusted-certs.pem
trust-store-password
```

**Comment by boaks** [comment]
Hm, I guess, that's a result of executing `create_certs.sh`.

**Comment by boaks** [comment]
> creating JKS trust store (certs/trustStore.jks) containing CA certificate
> ./create_certs.sh: Zeile 113: /bin/keytool: Datei oder Verzeichnis nicht gefunden
> ./create_certs.sh: Zeile 114: /bin/keytool: Datei oder Verzeichnis nicht gefunden

**Comment by boaks** [comment]
Looks like, `JAVA_HOME` is required. would it be possible, to replace the 

```
JAVA_KEY_TOOL=${JAVA_HOME}/bin/keytool
```

by 

```
if [ -z "${JAVA_HOME}" ]
then
   JAVA_KEY_TOOL=keytool
else
   JAVA_KEY_TOOL=${JAVA_HOME}/bin/keytool
fi
```

**Comment by sophokles73** [comment]
Looks like this would make the script more versatile. Would you mind creating a PR? I would prefer the first variant as it is much easier to understand for people who are not that deep into bash programming ...

**Comment by boaks** [comment]
I will prepare a PR in the late afternoon.

**Comment by boaks** [comment]
I was able to do it earlier.

**Comment by boaks** [comment]
Fixed with #3249
